app/api/upload/route.ts

```diff
@@ -0,0 +1,139 @@
+import { NextRequest, NextResponse } from 'next/server';
+import { createClient } from '@/utils/supabase/server';
+import { cookies } from 'next/headers';
+import {
+  getSecureUploadUrl,
+  getPublicFileUrl,
+  verifyFileExists,
+} from '@/services/server-file-upload';
+import { isValidExtension, isValidFileType } from '@/types/submission';
+
+/**
+ * API route for generating secure upload URLs
+ *
+ * Flow:
+ * 1. Client requests a signed upload URL with file info (type, name)
+ * 2. Server generates a secure URL with a short expiration
+ * 3. Client uploads directly to storage using the URL
+ * 4. Server verifies the file was uploaded and returns the public URL
+ */
+export async function POST(request: NextRequest) {
+  try {
+    // Check authentication
+    const cookieStore = cookies();
+    const supabase = createClient(cookieStore);
```

Comment:
There appear to be two issues with the Supabase client initialization:

1. The createClient() function doesn't take parameters (it handles cookie store internally), but you're passing cookieStore to it.

2. createClient() returns a Promise that needs to be awaited.

Suggested fix for both POST and GET handlers:
- Remove the separate cookieStore declaration 
- Change `const supabase = createClient(cookieStore)` to `const supabase = await createClient()`

This should resolve the TypeScript errors:
- "Expected 0 arguments, but got 1"
- "Property 'auth' does not exist on type 'Promise<SupabaseClient<any, "public", any>>'"